pkg/client/cluster-essentials/clusterinit.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package clusteressentials has APIs for bootstrapping kubernetes cluster
+package clusteressentials
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"k8s.io/client-go/rest"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/client/cluster-essentials/pkg/utils"
+)
+
+type ClusterEssentialsClient struct {
+	ClusterEssentialBundlePath string
+	BundleDestDir              string
+	RestConfig                 *rest.Config
+	BlockingCallEnable         bool
+	Calltimeout                time.Duration
+}
+
+var downloadBundle = utils.DownloadBundle
+
+func PreInstallValidation(restConfig *rest.Config) error {
+	// TODO implementation
+	return nil
+}
+
+func PostInstallValidation(restConfig *rest.Config, blockingEnable bool, timeout time.Duration) error {
+	// TODO implementation
+	return nil
+}
+
+func Install(c ClusterEssentialsClient) error {
+	err := PreInstallValidation(c.RestConfig)
+	if err != nil {
+		return fmt.Errorf("pre install validation is failed with error: %w", err)
+	}
+
+	if c.BundleDestDir == "" {
+		c.BundleDestDir = "tmp"
+	}
+
+	bundleDir, err := os.MkdirTemp(c.BundleDestDir, "cluster-essential")
+	if err != nil {
+		return err
+	}
+	err = downloadBundle(c.ClusterEssentialBundlePath, bundleDir)
+	if err != nil {
+		return fmt.Errorf("unable to download cluster essential manifest: %w", err)
+	}
+	packages, err := os.ReadDir(bundleDir)
+	if err != nil || len(packages) == 0 {
+		return fmt.Errorf("error while reading the bundleDir %s", bundleDir)
+	}
+	var generatedManifestBytes []byte
+	for _, item := range packages {
```

Comment:
We should only process directories within the bundleDir, just to be on the safe side.

Reply:
We are processing directories which are inside the bundledir , Please check L56